domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -989,79 +996,53 @@ public void saveBatch(String table, JsonArray entities, Handler<AsyncResult<Resu
+      JsonArray entities, Handler<AsyncResult<RowSet<Row>>> replyHandler) {
 
     try {
-      if (entities == null || entities.isEmpty()) {
-        saveBatchInternal(sqlConnection, table, entities, replyHandler);
-        return;
-      }
-      JsonArray idEntityPairs = new JsonArray();
-      for (int i = 0; i < entities.size(); i++) {
-        String json = entities.getString(i);
-        JsonObject jsonObject = new JsonObject(json);
-        String id = jsonObject.getString("id");
-        if (id == null) {
-          id = UUID.randomUUID().toString();
+      List<Tuple> list = new ArrayList<>();
+      if (entities != null) {
+        for (int i = 0; i < entities.size(); i++) {
+          String json = entities.getString(i);
+          JsonObject jsonObject = new JsonObject(json);
+          String id = jsonObject.getString("id");
+          list.add(Tuple.of(id == null ? UUID.randomUUID() : UUID.fromString(id),
+              jsonObject));
         }
-        idEntityPairs.add(id).add(json);
       }
-      saveBatchInternal(sqlConnection, table, idEntityPairs, replyHandler);
+      saveBatchInternal(sqlConnection, table, list, replyHandler);
     } catch (Exception e) {
       log.error(e.getMessage(), e);
       replyHandler.handle(Future.failedFuture(e));
     }
   }
 
-  /**
-   * Insert the entities into table using a single INSERT statement.
-   * @param sqlConnection  the connection to run on, may be on a transaction
-   * @param table  destination table to insert into
-   * @param idEntityPairs  for each enity the array contains two strings: the id as a string and the entity as a JSON string.
-   * @param replyHandler  result, containing the id field for each inserted entity
-   */
   private void saveBatchInternal(AsyncResult<SQLConnection> sqlConnection, String table,
-      JsonArray idEntityPairs, Handler<AsyncResult<ResultSet>> replyHandler) {
+                                  List<Tuple> batch, Handler<AsyncResult<RowSet<Row>>> replyHandler) {
 
     try {
       long start = System.nanoTime();
-      if (idEntityPairs == null || idEntityPairs.isEmpty()) {
-        // return empty result
-        ResultSet resultSet = new ResultSet(
-            Collections.singletonList(ID_FIELD), Collections.emptyList(), null);
-        replyHandler.handle(Future.succeededFuture(resultSet));
-        return;
-      }
-
-      log.info("starting: saveBatch size=" + idEntityPairs.size());
+      log.info("starting: saveBatch size=" + batch.size());
       StringBuilder sql = new StringBuilder()
           .append(INSERT_CLAUSE)
           .append(schemaName).append(DOT).append(table)
-          .append(" (id, jsonb) VALUES ");
-      for (int i = 0; i < idEntityPairs.size(); i += 2) {
-        if (i > 0) {
-          sql.append(',');
-        }
-        sql.append("(?,?)");
-      }
+          .append(" (id, jsonb) VALUES ($1, $2)");
       sql.append(RETURNING_ID);
```

Comment:
```suggestion
          .append(" (id, jsonb) VALUES ($1, $2)")
          .append(RETURNING_ID);
```

Reply:
Using a plain String. That should be fine for a one time assignment, IIRC.

Reply:
Yes.